### PR TITLE
refactor: disabled JSON-LD checks for VPs with JWT VCs by default

### PR DESCRIPTION
**Pull request by birtony**
Signed-off-by: Anton Biriukov <anton.biriukov@avast.com>

### CODE REVIEW

**Review by vkubiv — APPROVED** [review]

File: pkg/doc/verifiable/presentation.go [hunk 1]
```diff
@@ -9,15 +9,14 @@ import (
...
-	"github.com/xeipuuv/gojsonschema"
-
 	"github.com/hyperledger/aries-framework-go/pkg/doc/jose"
 	docjsonld "github.com/hyperledger/aries-framework-go/pkg/doc/jsonld"
 	"github.com/hyperledger/aries-framework-go/pkg/doc/jwt"
 	"github.com/hyperledger/aries-framework-go/pkg/doc/signature/verifier"
 	jsonutil "github.com/hyperledger/aries-framework-go/pkg/doc/util/json"
+	jsonld "github.com/piprate/json-gold/ld"
+	"github.com/xeipuuv/gojsonschema"
+	"strings"
```
Comment (Moopli) [inline pkg/doc/verifiable/presentation.go:19]: do we still have a linter check for grouping imports?
I noticed the linter didn't seem to catch these being in the wrong place.

Comment (DRK3) [reply pkg/doc/verifiable/presentation.go:19]: I noticed this too - I remember updating the linter version awhile back - I noticed in other repos that use the newer linter that it's not checking either. Maybe something with the new version

File: pkg/doc/verifiable/presentation.go [hunk 2]
```diff
@@ -402,6 +402,14 @@ func WithPresJSONLDDocumentLoader(documentLoader jsonld.DocumentLoader) Presenta
 	}
 }
 
+// WithPresRequiredVCJSONLDChecks requires performing JSON-LD checks for the VC when parsing a VP.
+// By default, JSON-LD checks are disabled for VPs with VCs that use a JWT format.
+func WithPresRequiredVCJSONLDChecks() PresentationOpt {
```
Comment (DRK3) [inline pkg/doc/verifiable/presentation.go:407]: Can simplify to `WithRequiredVCJSONLDChecks`

File: pkg/doc/verifiable/presentation.go [hunk 3]
```diff
@@ -542,12 +561,15 @@ func decodeCredentials(rawCred interface{}, opts *presentationOpts) ([]interface
...
 }
 
-func validateVP(data []byte, opts *presentationOpts) error {
+func validateVP(data []byte, opts *presentationOpts, isVPWithJWTCredential bool) error {
 	err := validateVPJSONSchema(data)
 	if err != nil {
 		return err
 	}
 
+	if isVPWithJWTCredential && !opts.requireVCJSONLDChecks {
```
[the remark this answers is not in the document]
Comment (birtony) [reply pkg/doc/verifiable/presentation.go]: As discussed, we'd rather let the consumer decide whether checks are needed or not by setting an optional parameter to disable them

**Review by DRK3 — APPROVED** [review]
Looks good to me, if it looks good to @Moopli then I think it can be merged

**Review by Moopli — APPROVED** [review]

File: pkg/doc/verifiable/presentation.go [hunk 4]
```diff
@@ -402,6 +403,14 @@ func WithPresJSONLDDocumentLoader(documentLoader jsonld.DocumentLoader) Presenta
 	}
 }
 
+// WithDisabledJSONLDChecks disables JSON-LD checks for VP parsing.
+// By default, JSON-LD checks are enabled.
+func WithDisabledJSONLDChecks() PresentationOpt {
```
Comment (sudeshrshetty) [inline pkg/doc/verifiable/presentation.go:408]: Please specify in comments atleast, what you mean by JSON LD checks, is it JSON LD Schema Validation?

**Review by sudeshrshetty — APPROVED** [review]

**Pull request merged**